tests/test_playbooks/tasks/inventory_plugin.yml

```diff
@@ -1,164 +1,13 @@
-
-- name: create test hosts
-  host:
-    username: admin
-    password: changeme
-    server_url: http://127.0.0.1:{{ foreman_port }}/
-    name: "{{ item.key }}"
-    hostgroup: "{{ item.value }}"
-    build: false
-    organization: Default Organization
-    location: Default Location
-    managed: false
-    parameters:
-      - name: testparam1
-        value: testvalue1
-      - name: testparam2
-        value: testvalue2
-  with_dict: "{{ foreman_hosts }}"
-
-- name: submit facts for hosts
-  uri:
-    url: http://127.0.0.1:{{ foreman_port }}/api/hosts/facts
-    url_username: admin
-    url_password: changeme
-    validate_certs: false
-    method: POST
-    body:
-      name: "{{ item.key }}"
-      facts:
-        fqdn: "{{ item.key }}"
-        domain: example.com
-        famtesthost: true
-        operatingsystem: CentOS
-        operatingsystemrelease: "7.7"
-    force_basic_auth: true
-    status_code: 201
-    body_format: json
-  with_dict: "{{ foreman_hosts }}"
-  ignore_errors: true
-
-- name: Refresh inventory to ensure new instances exist in inventory
-  meta: refresh_inventory
+- name: setup
+  include_tasks: inventory_plugin_setup.yml
 
 - name: execute tests
   block:
-    - name: test that all groups are present
-      assert:
-        that: >
-          'foreman_{{ item.label | default(item.name) | regex_replace('/', '_') }}' in groups
-      with_items: "{{ foreman_groups }}"
-
-    - name: test that all hosts are in the "all" group
-      assert:
-        that: >
-          '{{ item.key }}' in groups['all']
-      with_dict: "{{ foreman_hosts }}"
-
-    - name: test that all hosts are in the correct hostgroup
-      assert:
-        that: >
-          '{{ item.key }}' in groups['foreman_{{ item.value | regex_replace('/', '_') }}']
-      with_dict: "{{ foreman_hosts }}"
-
-    - name: dump all host details
-      debug:
-        var: "hostvars['{{ item.key }}']"
-      with_dict: "{{ foreman_hosts }}"
-
-    - name: test that all hosts have the domain fact set
-      assert:
-        that: >
-          hostvars['{{ item.key }}']['foreman_facts']['domain']  == 'example.com'
-      with_dict: "{{ foreman_hosts }}"
-
-    - name: test that all hosts have the OS fact set
-      assert:
-        that: >
-          hostvars['{{ item.key }}']['foreman_facts']['operatingsystem']  == 'CentOS'
-      with_dict: "{{ foreman_hosts }}"
-
-    - name: test that all hosts have the testparam1
-      assert:
-        that: >
-          hostvars['{{ item.key }}']['testparam1']  == 'testvalue1'
-      with_dict: "{{ foreman_hosts }}"
-
-    - name: test that all hosts have the testparam2
-      assert:
-        that: >
-          hostvars['{{ item.key }}']['testparam2']  == 'testvalue2'
-      with_dict: "{{ foreman_hosts }}"
+    - name: execute tests
+      include_tasks: inventory_plugin_tests.yml
+    - name: execute tests again, now fed from cache
+      include_tasks: inventory_plugin_tests.yml
```

Comment:
this doesn't seem to work… I would expect the facts asserts to fail for the report api codepath, but they do not.